### ISSUE TITLE
Filter by Tag when clicking on Tag is disorienting

### ISSUE THREAD

**Issue by privatemaker** [nextcloud/deck#5134]
**Is your feature request related to a problem?**

When a new Deck user first clicks on a Tag in a card the filter is applied to all cards on that list.

![deck-ux-click-filter](https://github.com/nextcloud/deck/assets/49612519/7c68ec93-2b7f-4d82-86b4-af36485bc655)

All other cards seemingly disappear very quickly and the only change to the UI is the `Unfiltered` icon changes to `Filter` 

![deck-ux-click-unfilter](https://github.com/nextcloud/deck/assets/49612519/b3de6c5b-6a6b-4000-b1ed-f6971036fd2d)

Due to the abruptness of this and the fact there is only one very small detail in the UI which changes, is easy to not notice or understand what happened.

This creates a feeling of discomfort and concern that data has been deleted or such.

**Describe the solution you'd like**

Show a stronger UI presence that filters have been applied to this view.

Perhaps a word next to the icon + a counter, like:

```
[Icon] No Filters
[Icon] 2 Filters
```

**Describe alternatives you've considered**

Visually showing Tag filters (limit 3 or so) with colored background in the topbar, perhaps on the left of the search field.

**Comment by juliushaertl** [comment]
Thanks for the suggestion. We don't use counters for this anywhere in Nextcloud as far as I'm aware.

Currently we switch the icon depending on if filters are set or not:

- No filter active ![Screenshot 2023-10-10 at 09 17 30](https://github.com/nextcloud/deck/assets/3404133/fc8f122d-cc0a-4d2f-bc34-9bdbd19ab33f)
- Filter(s) active ![Screenshot 2023-10-10 at 09 17 35](https://github.com/nextcloud/deck/assets/3404133/2b4e623c-673d-4d49-a72d-4380bdd22948)

However I see that this is not enough and can easily be overseen.

Talk indicates active filters with a primary color button style, so maybe we should do the same:
<img width="236" alt="Screenshot 2023-10-10 at 09 18 27" src="https://github.com/nextcloud/deck/assets/3404133/cf324308-ec4e-4de7-acd3-f9689afd330f">

**Comment by fitrahfm** [comment]
Hi @privatemaker @juliushaertl I would love to work on this

**Comment by juliushaertl** [comment]
Feel free to tackle this. Pull requests are always welcome. If you have any question don't hesitate to reach out :)

**Comment by privatemaker** [comment]
> We don't use counters for this anywhere in Nextcloud as far as I'm aware.

Is there a strong reason for this other than uniformity? In certain cases (like Filters) it's really helpful and a common industry standard to communicate state of things to a user. Take the counters in Github's repo tab bar which are essential UX:

![ux-research-github-number-counters](https://github.com/nextcloud/deck/assets/49612519/d5c0b660-4c8e-4106-bc52-b42da517cf4a)

> Talk indicates active filters with a primary color button style, so maybe we should do the same:

@juliushaertl I'm not sure what you mean, I don't see any filters in the instance of Talk I use. Maybe post screenshot?

cc: @nextcloud/designers

**Comment by juliushaertl** [comment]
> Is there a strong reason for this other than uniformity? In certain cases (like Filters) it's really helpful and a common industry standard to communicate state of things to a user. Take the counters in Github's repo tab bar which are essential UX

For the active filters I think a bare number does not give any additional information that a clear indication that "some" filters are active wouln't already give.

>> Talk indicates active filters with a primary color button style, so maybe we should do the same:

> @juliushaertl I'm not sure what you mean, I don't see any filters in the instance of Talk I use. Maybe post screenshot?

Sure, I was referring to this button which is what #5299 implements:

<img width="282" alt="Screenshot 2023-11-20 at 07 38 48" src="https://github.com/nextcloud/deck/assets/3404133/908e1b1e-08c7-4cf9-b136-46296af3ab87">


<img width="362" alt="Screenshot 2023-11-20 at 07 38 02" src="https://github.com/nextcloud/deck/assets/3404133/1cb66790-2a06-4270-bf2d-ee989ff4d0bd">

**Comment by privatemaker** [comment]
>  I think a bare number does not give any additional information that a clear indication that "some" filters are active wouldn't already give.

While I agree that a bare number is not very meaningful, what I suggested after considering what is best is the following:

```
[Icon] No Filters
[Icon] 2 Filters
```

Thanks for the screenshot from Talk. That must be a newer version from what I regularly use, given the differences of Talk vs. Deck apps being quite different I don't think it will have same jarring effect I describe.

What I currently see in Deck on my Nextcloud instance:

![deck-filter-list-tooltip](https://github.com/nextcloud/deck/assets/49612519/473e65bb-03f3-4aa9-bdae-b37b31635128)

From the screenshot posted in #5299 it appears to only add a gray circle around the icon. That is not nearly strong enough visual change or communication to address the issue in my opinion.

**Comment by nimishavijay** [comment]
Agreed that we could use a more prominent indicator when there are filters applied, here's what trello does:

Before filters have been applied:

![image](https://github.com/nextcloud/deck/assets/52440189/803044c5-80a1-4753-b34e-5370ab93fce0)

After filters have been applied:
![image](https://github.com/nextcloud/deck/assets/52440189/419597c1-15ed-4933-aa5d-45d0b41a1638)

- The "filter section" is highlighted with a background color 
- In the filter section there is an indication of how many cards match the filter (this was super confusing tbh, I kept thinking that this number was the number of filters applied, but for people who are trello power users this may be expected? 🤷)
- In the filter section there is also a "clear all" button
- There is a subline for each column title indicating how many cards match the filter

How about taking some of these features and adapting it to Deck? :) 

- [ ] highlighting the filter icon in the style of Talk 
  - consistent with our style like @juliushaertl mentioned
  - also would make sure that the position of other elements in that row don't shift depending on if filters are applied 
- [ ] adding a subline to the column title indicating how many cards match the filter
  - "2 cards match filter" so an explicit message like @privatemaker suggested
  - not in the corner of the screen, so users of big screens will have another indication in case the filter button color change is missed

What do you think? :) sort of a combination of both your ideas @juliushaertl @privatemaker :)
